server/src/main/java/de/uftos/repositories/ucdl/UcdlRepositoryImpl.java

```diff
@@ -11,24 +17,63 @@
  */
 @Repository
 public class UcdlRepositoryImpl implements UcdlRepository {
+  private final File ucdlFile = new File("/app/ucdl/ucdl.yml");
+  private HashMap<String, ConstraintDefinitionDto> currentDefinitions = null;
 
   @Override
   public String getUcdl() {
-    return "";
+    try {
+      ucdlFile.createNewFile();
+      FileReader reader = new FileReader(ucdlFile);
+      int readInformation = reader.read();
+      StringBuilder sb = new StringBuilder();
+      while (readInformation >= 0) {
+        sb.append((char) readInformation);
+        readInformation = reader.read();
+      }
+      return sb.toString();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
   }
 
   @Override
   public void setUcdl(String ucdl) {
+    this.currentDefinitions = null;
+    try {
+      FileWriter writer = new FileWriter(ucdlFile);
+      writer.write(ucdl);
+      writer.close();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
 
   }
 
   @Override
   public ParsingResponse parseFile() {
-    return null;
+    try {
+      this.setCurrentDefinitions();
+    } catch (ParseException | IOException e) {
+      return new ParsingResponse(false, e.getMessage());
+    }
+    return new ParsingResponse(true, "Parsing was successful!");
   }
 
   @Override
-  public HashMap<String, ConstraintDefinitionDto> getConstraints() throws ParseException {
-    return null;
+  public HashMap<String, ConstraintDefinitionDto> getConstraints() {
```

Comment:
You could simplify this whole function to
```java
if (this.currentDefinitions == null && !this.parseFile().success()) {
	return null;
}
return this.currentDefinitions;
```
Then you can get rid of the `setCurrentDefinitions()` function as well